Shoko.Server/API/v3/Controllers/RenamerController.cs

```diff
@@ -13,39 +14,204 @@ namespace Shoko.Server.API.v3.Controllers
+
+            var script = RepoFactory.RenameScript.GetByName(body.Name);
+            if (script != null)
+                return BadRequest("A script with the given name already exists!");
+
+            var renamer = RenameFileHelper.Renamers.Values.FirstOrDefault(r => r.ID == body.IDs.Renamer.Value);
+            if (renamer == null)
+                return BadRequest("Renamer not found.");
+
+            script = new Shoko.Models.Server.RenameScript
+            {
+                ScriptName = body.Name,
+                RenamerType = renamer.Name,
+                IsEnabledOnImport = body.EnabledOnImport ? 1 : 0,
+                Script = body.Body,
+                ExtraData = null,
+            };
+            RepoFactory.RenameScript.Save(script);
+
+            return Ok();
+        }
+
+        /// <summary>
+        /// Get a <see cref="Models.Shoko.Renamer.Script"/> by the given <paramref name="scriptID"/>.
+        /// </summary>
+        /// <param name="scriptID">Script ID</param>
+        /// <returns>The script</returns>
+        [HttpGet("Script/{scriptID}")]
+        public ActionResult<Models.Shoko.Renamer.Script> GetRenamerScriptByScriptID([FromRoute] int scriptID)
+        {
+            var script = RepoFactory.RenameScript.GetByID(scriptID);
+            if (script == null)
+                return NotFound("Renamer.Script not found.");
+
+            return new Models.Shoko.Renamer.Script(script);
+        }
+
+        /// <summary>
+        /// Replace an existing <see cref="Models.Shoko.Renamer.Script"/> by the given <paramref name="scriptID"/>.
+        /// </summary>
+        /// <param name="scriptID">Script ID</param>
+        /// <param name="body">The new script to replace the entry with.</param>
+        /// <returns></returns>
+        [Authorize("admin")]
+        [HttpPut("Script/{scriptID}")]
+        public ActionResult RutRenamerScriptByScriptID([FromRoute] int scriptID, [FromBody] Models.Shoko.Renamer.Script body)
+        {
+            var script = RepoFactory.RenameScript.GetByID(scriptID);
+            if (script == null)
+                return NotFound("Renamer.Script not found.");
+
+            body.MergeWithExisting(script);
+            RepoFactory.RenameScript.Save(script);
+        
+            return Ok();
+        }
+
+        /// <summary>
+        /// Delete an existing <see cref="Models.Shoko.Renamer.Script"/> by the given <paramref name="scriptID"/>
+        /// </summary>
+        /// <param name="scriptID">Script ID</param>
+        /// <returns></returns>
+        [Authorize("admin")]
+        [HttpDelete("Script/{scriptID}")]
+        public ActionResult DeleteRenamerScriptByScriptID([FromRoute] int scriptID)
+        {
+            var script = RepoFactory.RenameScript.GetByID(scriptID);
+            if (script == null)
+                return NotFound("Renamer.Script not found.");
```

Comment:
```suggestion
                return NotFound("Renamer Script not found.");
```

Reply:
It's intentional. The model name is "Renamer.Script" in the api.